### PR TITLE
Rename nl_template_*_multi_line -> nl_template_*

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Remove the `_multi_line` from the recently added `nl_template_*` options. Other `*_multi_line` options are used for code that already spans multiple lines, but these options apply always; the `_multi_line` here is confusing and implies that the options work differently from how they actually work. Also, fix some issues with the option descriptions (namely, the descriptions mentioned parentheses, but actually apply to angle brackets).

### CODE REVIEW

File: src/options.h [hunk 1]
```diff
@@ -2118,30 +2118,33 @@ nl_func_call_empty;
...
-// Whether to add a newline after '(' in a function call if '(' and ')' are in
-// different lines.
+// Whether to add a newline after '(' in a function call
+// if '(' and ')' are in different lines.
 extern Option<bool>
 nl_func_call_start_multi_line;
 
-// Whether to add a newline after '(' in a template.
+// Whether to add a newline after '<' of a template parameter list
+// if '<' and '>' are in different lines.
```
Comment (guy-maurel) [inline src/options.h]: NO, even in the same line, see the test

File: src/options.h [hunk 2]
```diff
@@ -2118,30 +2118,33 @@ nl_func_call_empty;
...
-// Whether to add a newline after each ',' in a function call if '(' and ')'
-// are in different lines.
+// Whether to add a newline after each ',' in a function call
+// if '(' and ')' are in different lines.
 extern Option<bool>
 nl_func_call_args_multi_line;
 
-// Whether to add a newline after each ',' in a template.
+// Whether to add a newline after each ',' in a template parameter list
+// if '<' and '>' are in different lines.
```
Comment (guy-maurel) [inline src/options.h]: NO, even in the same line, see the test

File: src/options.h [hunk 3]
```diff
@@ -2118,30 +2118,33 @@ nl_func_call_empty;
...
-// Whether to add a newline before ')' in a function call if '(' and ')' are in
-// different lines.
+// Whether to add a newline before ')' in a function call
+// if '(' and ')' are in different lines.
 extern Option<bool>
 nl_func_call_end_multi_line;
 
-// Whether to add a newline before ')' in a template.
+// Whether to add a newline before '>' of a template parameter list
+// if '<' and '>' are in different lines.
```
Comment (guy-maurel) [inline src/options.h]: NO, even in the same line, see the test

File: src/options.h (hunk 1 above)
Comment (mwoehlke-kitware) [reply src/options.h]: Hmm, but if that's the case, shouldn't the option name just be `nl_template_start`?

Comment (gmaurel) [reply src/options.h]: Yes

**Comment by mwoehlke-kitware** [comment]
Is it right now?

**Pull request merged**